### ISSUE TITLE
why calculated sigmoid twice？

### ISSUE THREAD

**Issue by zpyovo** [VisionLearningGroup/DA_Detection#28]
Hello，in code vgg16_global_local.py，sigmoid has been calculated once，why count again in net_utils.py class FocalLoss(nn.Module)？
`class netD_pixel(nn.Module):
    def forward(self, x):
        # x = ReverseLayerF.apply(x, self.beta)
        x = F.relu(x)
        x = F.relu(self.conv1(x))
        x = F.relu(self.conv2(x))
        if self.context:
            feat = F.avg_pool2d(x, (x.size(2), x.size(3)))
            # feat = x
            **x = torch.sigmoid(self.conv3(x))**
            return x.view(-1,1), feat  # torch.cat((feat1,feat2),1)#F
        else:
            **x = torch.sigmoid(self.conv3(x))**
            return x.view(-1,1)#F.sigmoid(x)`
`class FocalLoss(nn.Module)：
 def forward(self, inputs, targets):
        N = inputs.size(0)
        # print(N)
        C = inputs.size(1)
        if self.sigmoid:
            **P = F.sigmoid(inputs)**
            #F.softmax(inputs)
            if targets == 0:
                probs = 1 - P#(P * class_mask).sum(1).view(-1, 1)
                log_p = probs.log()
                batch_loss = - (torch.pow((1 - probs), self.gamma)) * log_p
            if targets == 1:
                probs = P  # (P * class_mask).sum(1).view(-1, 1)
                log_p = probs.log()
                batch_loss = - (torch.pow((1 - probs), self.gamma)) * log_p
        else:
            #inputs = F.sigmoid(inputs)
            **P = torch.softmax(inputs)`**
